File: spoj/codeforces/ed82/2.py
for _ in range(int(input())):
    n,g,b = list(map(int,input().split()))
    if g>=b:
        print(n)
        continue
    ans = 0
    m = n
    n=(n+1)//2
    com = n//g
    if n%g != 0:
        print(max(m,com*(b+g)+n%g))
    else:
        print(max(m,com*(b+g)-b))

File: spoj/hackerrank/pir_2/2.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){

    int n;cin>>n;
    vector<int>arr;
    arr.push_back(0);
    for(int i=0;i<n;i++){
        int x;cin>>x;
//         if(!arr.size()){
//             arr.push_back(x);
//             continue;
//         }
        arr.push_back(arr.back()^x);
    }
    int x;cin>>x;
//     int ans = -1;
//     for(int i=0;i<=n;i++){
//         int req = x^arr[i];
// 
//     }
    map<int,vector<int> > mp;
    for(int i=0;i<=n;i++){
        mp[arr[i]].push_back(i);
    }
    int ans = -1;
    for(int i=0;i<=n;i++){
        int req = arr[i]^x;
        if(mp[req].size() > 0){
            ans = max(ans,mp[req].back()-i);
        }
    }
    cout<<ans<<endl;
}
